Tools/package_client_build.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python3
+
+import os
+import shutil
+import subprocess
+import sys
+import zipfile
+import argparse
+
+from typing import List, Optional
+
+try:
+    from colorama import init, Fore, Style
+    init()
+
+except ImportError:
+    # Just give an empty string for everything, no colored logging.
+    class ColorDummy(object):
+        def __getattr__(self, name):
+            return ""
+
+    Fore = ColorDummy()
+    Style = ColorDummy()
+
+
+p = os.path.join
+
+SHARED_IGNORED_RESOURCES = {
+    "ss13model.7z",
+    "ResourcePack.zip",
+    "buildResourcePack.py",
+    "CONTENT_GOES_HERE",
+    ".gitignore",
+    ".directory",
+    ".DS_Store"
+}
+
+CLIENT_IGNORED_RESOURCES = {
+    "Maps",
+    "emotes.xml",
+    "Groups",
+    "engineCommandPerms.yml",
+    "clientCommandPerms.yml"
+}
+
+CLIENT_CONTENT_ASSEMBLIES = [
+    "OpenDreamClient",
+    "OpenDreamShared"
+]
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="Packages the SS14 client.")
+
+    parser.add_argument("--skip-build",
+                        action="store_true",
+                        help=argparse.SUPPRESS)
+
+    args = parser.parse_args()
+    skip_build = args.skip_build
+
+    if os.path.exists("release"):
+        pass
+    #    print(Fore.BLUE + Style.DIM +
+    #          "Cleaning old release packages (release/)..." + Style.RESET_ALL)
+    #    shutil.rmtree("release")
+    else:
+        os.mkdir("release")
+
+    if not skip_build:
+        wipe_bin()
+
+    build(skip_build)
+
+
+def wipe_bin():
+    print(Fore.BLUE + Style.DIM +
+          "Clearing old build artifacts (if any)..." + Style.RESET_ALL)
+
+    if os.path.exists("bin"):
+        shutil.rmtree("bin")
+
+# Be advised this is called from package_server_build during a Hybrid-ACZ build.
+def build(skip_build: bool) -> None:
+    # Run a full build.
+    print(Fore.GREEN + "Building project..." + Style.RESET_ALL)
+
+    if not skip_build:
+        subprocess.run([
+            "dotnet",
+            "build",
+            p("OpenDreamClient", "OpenDreamClient.csproj"),
+            "-c", "Release",
+            "--nologo",
+            "/v:m",
+            "/t:Rebuild",
+            "/p:FullRelease=True",
+            "/m"
+        ], check=True)
+
+    print(Fore.GREEN + "Packaging client..." + Style.RESET_ALL)
+
+    client_zip = zipfile.ZipFile(
+        p("release", "SS14.Client.zip"), "w",
```

Comment:
Would changing the name of this zip break anything?

Reply:
Yes, and I learned that the hard way. I made an issue already on the SS14 repo to debrand the tools.